CMSgov/onemac
# OY2-17028: Add "Proposed Effective Date" to CHIP and Medicaid SPA details pages (not RAI)

Story: https://qmacbis.atlassian.net/browse/OY2-17028
Endpoint: https://d1y5t3v76zj4uo.cloudfront.net/

### Details
Follow the form pattern and split out the Details pages into the different types.
Also, add the Proposed Effective Date field to Medicaid and CHIP SPA detail pages.

### Changes
- Reorganized the Details pages so that the configuration "wraps" the component
- created the specific pages for Medicaid SPA, CHIP SPA, Base Waiver, Temporary Extension, and Waiver Amendment
- modified the routing so that we default to "page not found" for any pages not touched (was using the default DetailView and causing confusion about what pages were "done").
- Split Unit Tests into new structure
- added a waiver amendment to the seed data to support the Amendment Details page (story: https://qmacbis.atlassian.net/browse/OY2-15971)

### Test Plan
1. Login as a submitting user
2. Navigate to Packages Dashboard - SPA Tab
3. Click on a Medicaid SPA ID link
4. Verify the Details page looks as expected
5. Verify Proposed Effective Date is in the details
6. Return to Packages Dashboard - SPA Tab
7. Click on a CHIP SPA ID link
8. Verify the Details page looks as expected
9. Verify Proposed Effective Date is in the details
10. Return to Packages Dashboard - Waiver Tab
11. Click on a Base Waiver ID link
12. Verify the Details page looks as expected
13. Return to Packages Dashboard - Waiver Tab
14. Click on a Waiver Amendment ID link
15. Verify the Details page looks as expected
16. Return to Packages Dashboard - Waiver Tab
17. Click on a Temporary Extension ID link
18. Verify the Details page looks as expected
19. Return to Packages Dashboard
